Modules/_cursesmodule.c

```diff
@@ -4944,9 +5042,24 @@ curses_capi_capsule_new(void *capi)
 
 /* Module initialization */
 
+/* Indicate whether the module has been loaded or not. */
+static int curses_module_loaded = 0;
+
 static int
 cursesmodule_exec(PyObject *module)
 {
+    if (_Py_atomic_load_int(&curses_module_loaded)) {
```

Comment:
The atomics here don't really do anything useful because you are not atomically checking and setting the variable together.  Either use an atomic compare-exchange or don't use atomics at all. 

Within an interpreter, the import lock will prevent multiple concurrent executions of `cursesmodule_exec`, but I'm not sure about what happens with multiple interpreters.

Reply:
> Within an interpreter, the import lock will prevent multiple concurrent executions of cursesmodule_exec

Oh ok. Thanks for the tip. I wasn't really sure on this one. I'll revert this change later